nms/app/views/domain-proxy/CbsdEdit.js

```diff
@@ -0,0 +1,425 @@
+    background: colors.primary.comet,
+    fontFamily: typography.button.fontFamily,
+    fontWeight: typography.button.fontWeight,
+    fontSize: typography.button.fontSize,
+    lineHeight: typography.button.lineHeight,
+    letterSpacing: typography.button.letterSpacing,
+
+    '&:hover': {
+      background: colors.primary.mirage,
+    },
+  },
+  tabBar: {
+    backgroundColor: colors.primary.brightGray,
+    color: colors.primary.white,
+  },
+  frequencyPreferencesContent: {
+    marginTop: theme.spacing(1),
+  },
+  bandwidthTitleContainer: {
+    height: theme.spacing(4),
+  },
+  frequencyPreferencesIcon: {
+    padding: 0,
+  },
+  addFrequencyIconContainer: {
+    textAlign: 'right',
+  },
+}));
+
+type ButtonProps = {
+  title: string,
+  isLink: boolean,
+};
+
+export function AddEditCbsdButton(props: ButtonProps) {
+  const classes = useStyles();
+  const [open, setOpen] = useState(false);
+
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+
+  return (
+    <>
+      <CbsdAddEditDialog open={open} onClose={handleClose} />
+      {props.isLink ? (
+        <Button
+          component="button"
+          variant="text"
+          onClick={handleOpen}>
+          {props.title}
+        </Button>
+      ) : (
+        <Button
+          variant="text"
+          className={classes.appBarBtn}
+          onClick={handleOpen}>
+          {props.title}
+        </Button>
+      )}
+    </>
+  );
+}
+
+type DialogProps = {
+  open: boolean,
+  onClose: () => void,
+  cbsd?: cbsd,
+};
+
+type BandwidthEnum = 5 | 10 | 15 | 20;
+
+type CbsdFormData = {
+  serialNumber: string,
+  fccId: string,
+  userId: string,
+  minPower: number,
+  maxPower: number,
+  numberOfAntennas: number,
+  antennaGain: number,
+  bandwidthMhz: BandwidthEnum,
+  frequenciesMhz: number[],
+};
+
+const convertToCbsdFormData = (cbsd?: cbsd): CbsdFormData => {
+  return {
+    serialNumber: cbsd?.serial_number || '',
+    fccId: cbsd?.fcc_id || '',
+    userId: cbsd?.user_id || '',
+    minPower: cbsd?.capabilities?.min_power || 0,
+    maxPower: cbsd?.capabilities?.max_power || 0,
+    numberOfAntennas: cbsd?.capabilities?.number_of_antennas || 0,
+    antennaGain: cbsd?.capabilities?.antenna_gain || 0,
+    bandwidthMhz: cbsd?.frequency_preferences?.bandwidth_mhz || 5,
+    frequenciesMhz: cbsd?.frequency_preferences?.frequencies_mhz || [0],
+  }
+}
+
+export function CbsdAddEditDialog(props: DialogProps) {
+  const enqueueSnackbar = useEnqueueSnackbar();
+
+  const classes = useStyles();
+
+  const ctx = useContext(CbsdContext);
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [formErrors, setFormErrors] = useState < string[] > ([]);
+  const [cbsdFormData, setCbsdFormData] = useState < CbsdFormData > (convertToCbsdFormData());
+
+  useEffect(() => {
+    const newValue = convertToCbsdFormData(props?.cbsd);
+    setCbsdFormData(newValue);
+  }, [props.open, props.cbsd]);
+
+  useEffect(() => {
```

Comment:
Maybe I'm missing something here. What is the purpose of this effect? `formErrors` are local state anyhow.

Reply:
I see the intention was probably have `props.open` in the dependencies to rest the error on close.

Reply:
Fixed